main.tf

```diff
@@ -315,15 +315,40 @@ data "aws_iam_policy_document" "main" {
     actions   = ["s3:GetBucketAcl"]
     resources = [local.bucket_arn]
   }
-
 }
 
+data "aws_iam_policy_document" "tls_requests_only" {
```

Comment:
I think it makes sense to just enforce TLS requests always for this module, so you can merge this with the main IAM policy document. I would expect every AWS service logging to this bucket should use TLS and if they don't we have bigger problems. 

Tests should should be able to confirm if this works, because each AWS service writes a test file to the bucket to confirm functionality.

Reply:
Grand; I'll change the default behaviour and remove the additional variable.

I was working from the assumption that modifying the default behaviour would be less preferable, but it's certainly a lot simpler to augment the current policy as you say.

My concern wouldn't be that AWS services would have any issue with TLS-only, but I've got no way to reason about whatever tooling folk are using to read log data that's been written in.